src/components/Taxonomy/Taxonomy.tsx

```diff
@@ -148,9 +160,37 @@ function isSubArray(item: string[], parent: string[]) {
   return parent.every((n, i) => item[i] === n);
 }
 
+type HintTooltipProps = {
+  // Without title there is no tooltip at all as a component
+  title?: string,
+  // wrapper is used as a tag in JSX to wrap child elements to make Tooltip to work with the single child element
+  // it can be a real tag or a component that provides real HTMLElement (not a text) as the result
+  wrapper?: CNTagName,
+  children: JSX.Element,
+}
+
+export const HintTooltip: React.FC<HintTooltipProps>  = ({
+  title,
+  wrapper: Wrapper, children,
```

Comment:
split this up so its a little more readable
```suggestion
  wrapper: Wrapper,
  children,
```